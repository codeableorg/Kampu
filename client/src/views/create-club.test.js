import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CreateClubfrom from "./create-club";

test("CreateClub View", () => {
  const { asFragment, getByLabelText } = render(<CreateClubfrom />);

  expect(asFragment()).toMatchSnapshot();

  const inputName = getByLabelText("enter name");
  fireEvent.change(inputName, {
    target: { value: "Club green" }
  });

  const inputAddress = getByLabelText("enter address");
  fireEvent.change(inputAddress, {
    target: { value: "Jr narnia 404" }
  });

  const file = new File(["(⌐□_□)"], "shinji-ikari.png", { type: "image/png" });
  const imageInput = getByLabelText("choose image");
  fireEvent.change(imageInput, { target: { files: [file] } });

  fireEvent.submit(inputName);
});
