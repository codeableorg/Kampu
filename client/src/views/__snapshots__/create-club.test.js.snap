// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateClub View 1`] = `
<DocumentFragment>
  <div
    class="css-1ha532s-CreateClub"
  >
    <form>
      <div
        class="css-5ossh7-CreateClub"
      >
        <label
          class="css-1vx6px3-Label"
          for="name"
        >
          Name
        </label>
        <input
          aria-label="enter name"
          autocomplete="off"
          class="css-s3z14w-Input"
          id="name"
          name="name"
          placeholder="Club's name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="css-5ossh7-CreateClub"
      >
        <label
          class="css-1vx6px3-Label"
          for="address"
        >
          Address
        </label>
        <input
          aria-label="enter address"
          autocomplete="off"
          class="css-s3z14w-Input"
          id="address"
          name="address"
          placeholder="Club's address"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="css-5ossh7-CreateClub"
      >
        <label
          class="css-1vx6px3-Label"
          for="image"
        >
          Image(s)
        </label>
        <input
          aria-label="choose image"
          class="css-s3z14w-Input"
          id="image"
          name="image"
          required=""
          type="file"
        />
      </div>
      <div
        class="css-5ossh7-CreateClub"
      >
        <label
          class="css-1vx6px3-Label"
          for="schedule"
        >
          Schedule
        </label>
        <input
          class="css-s3z14w-Input"
          id="schedule"
          name="schedule"
          type="text"
          value="{\\"monday-friday\\":{\\"init\\":\\"8\\",\\"end\\":\\"22\\"},\\"saturday\\":{\\"init\\":\\"8\\",\\"end\\":\\"22\\"},\\"sunday\\":{\\"init\\":\\"8\\",\\"end\\":\\"22\\"}}"
        />
      </div>
      <div
        class="css-5ossh7-CreateClub"
      >
        <button
          class="css-lrasnk-Button"
        >
          Create
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
